indexing-hadoop/src/main/java/org/apache/druid/indexer/DeterminePartitionsJob.java

```diff
@@ -437,13 +458,18 @@ protected void innerMap(
   static class DeterminePartitionsDimSelectionMapperHelper
   {
     private final HadoopDruidIndexerConfig config;
-    private final String partitionDimension;
+    private final List<List<String>> dimensionGroupingSet;
     private final Map<Long, Integer> intervalIndexes;
+    private final boolean supportNullValue;
 
-    DeterminePartitionsDimSelectionMapperHelper(HadoopDruidIndexerConfig config, String partitionDimension)
+    DeterminePartitionsDimSelectionMapperHelper(HadoopDruidIndexerConfig config)
     {
       this.config = config;
-      this.partitionDimension = partitionDimension;
+
+      DimensionRangePartitionsSpec spec = (DimensionRangePartitionsSpec) config.getPartitionsSpec();
+      final DimensionsSpec dimensionsSpec = config.getSchema().getDataSchema().getDimensionsSpec();
+      this.dimensionGroupingSet = new ArrayList<>(spec.getDimensionGroupingSet(dimensionsSpec));
+      this.supportNullValue = spec.supportNullValue();
```

Comment:
Hi @kfaraz, the single_dim does not support null values before, but actually it can. Shall I remove this and just let the range and single_dim both support null values?

Reply:
Yeah, that would be okay too.